pages/understanding-json-schema/reference/non_json_data.md

```diff
@@ -7,84 +7,128 @@ section: docs
-    string as-is. This includes text-based content types, such as
-    `text/html` or `application/xml`.
-- If the content is binary data, set `contentEncoding` to `base64` and
-    encode the contents using
-    [Base64](https://tools.ietf.org/html/rfc4648). This would include
-    many image types, such as `image/png` or audio types, such as
-    `audio/mpeg`.
+The acceptable values are:
+- `quoted-printable`
+- `base16` 
+- `base32` 
+- `base64` 
+ 
+If not specified, the encoding is the same as the containing JSON document.
 
-<Keywords label="single: contentSchema single: media; contentSchema" />
+For modern usage, there are two main scenarios:
 
-## contentSchema
-<Star label="New in draft 2019-09" />
+1. **Same encoding as JSON document**: Leave `contentEncoding` unspecified and include the content in a string as-is. This is suitable for text-based content types (e.g., `text/html`, `application/xml`) and assumes UTF-8 encoding in most cases.
+2. **Binary data**: Set `contentEncoding` to `base64` and encode the content using Base64. This is appropriate for binary content types such as images (`image/png`) or audio files (`audio/mpeg`).
 
-Documentation Coming soon
 
-## Examples
+Example:
 
-The following schema indicates the string contains an HTML document,
-encoded using the same encoding as the surrounding document:
+The following schema indicates that a string contains a PNG file and is encoded using Base64:
 
 ```json
 // props { "isSchema": true }
 {
   "type": "string",
-  "contentMediaType": "text/html"
+  "contentEncoding": "base64",
+  "contentMediaType": "image/png"
 }
 ```
 ```json
 // props { "indent": true, "valid": true }
-"<!DOCTYPE html><html xmlns=\"http://www.w3.org/1999/xhtml\"><head></head></html>"
+"iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAABmJLR0QA/wD/AP+gvaeTAAAA..."
 ```
-The following schema indicates that a string contains a PNG image, encoded using Base64:
+
+To better understand how `contentEncoding` and `contentMediaType` are applied in practice, let's consider the process of transmitting non-JSON data:
+
+![Role of contentEncoding and contenMediaType keywords in the transmission of non-JSON data](/media-keywords.png)
+
+
+1. The sender encodes the content, using `contentEncoding` to specify the encoding method (e.g., base64) and `contentMediaType` to indicate the media type of the original content.
+2. The encoded data is then transmitted.
+3. Upon receiving the data, the consumer application uses the `contentEncoding` and `contentMediaType` information to select the appropriate decoding method.
+4. Finally, the consumer application decodes the data, restoring it to its original form.
+
+This process ensures that the non-JSON content is properly encoded for transmission and accurately decoded by the recipient, maintaining the integrity of the data throughout the processe.
+
+<Keywords label="single: contentSchema single: media; contentSchema" />
+
```

Comment:
Based on the content of the contributing.md file and the Google style guide, I has just assumed that we are replacing the <keywords>, <star> tags in the new documentation. Please let me know in case my assumption is incorrect. Thank you

Reply:
The goal of the review and editing work is to make sure that the documentation uses the tags according to the rules documented in the style guide: https://github.com/json-schema-org/website/blob/main/pages/md-style-guide.md
That could mean replacing the <Star/> tag when it doesn't follow the guidelines with another tag best suited for the context.  

Regarding the <Keywords/> tag, we don't have it documented in the style guide, yet. But that is a topic for another conversation that I posted on the general channel.